lib/App/index.js

```diff
@@ -1581,254 +1570,202 @@ class App {
-            message: 'Add more arguments?'
-          }
-        ]);
+    if (flowArguments.length) {
+      flowJson.args = flowArguments;
+    }
 
-        // Create a custom object to inject en flag for the placeholder.
-        flowArgs.push({
-          type: argumentAnswers.type,
-          name: argumentAnswers.name,
-          placeholder: { en: argumentPlaceholder },
-        });
+    if (flowTokens.length) {
+      flowJson.tokens = flowTokens;
+    }
 
-        if( addMore.more ) {
-          await addArgument();
-        } else {
-          return;
-        }
-      }
+    return flowJson;
+  }
 
-      await addArgument();
-    }
+  static async _questionFlowArguments() {
+    const flowArguments = [];
 
-    const useTokens = await inquirer.prompt([
+    let { addArgument } = await inquirer.prompt([
       {
         type: 'confirm',
-        name: 'using_tokens',
-        message: 'Do you want to use tokens for this Flow Card?',
+        name: 'addArgument',
+        message: 'Do you want to use arguments for this Flow Card?',
         default: false
       }
     ]);
 
-    let flowTokens = [];
-    if( useTokens.using_tokens ) {
-      async function addToken() {
-        const { tokenTitle, tokenExample } = await inquirer.prompt([
-          {
-            type: 'input',
-            name: 'tokenTitle',
-            message: 'Enter the user title of your token',
-            validate: input => input.length > 0,
-          },
-          {
-            type: 'input',
-            name: 'tokenExample',
-            message: 'Give a brief example of what your token can provide',
-            validate: input => input.length > 0,
-          },
-        ]);
-
-        let tokenAnswers =  await inquirer.prompt([
-          {
-            type: 'list',
-            name: 'type',
-            message: 'What is the type of the token?',
-            choices: () => {
-              return [
-                {
-                  name: "Text",
-                  value: "string"
-                },
-                {
-                  name: "Number",
-                  value: "number"
-                },
-                {
-                  name: "Boolean",
-                  value: "boolean"
-                },
-                {
-                  name: "Image",
-                  value: "image"
-                }
-              ]
-            }
-          },
-          {
-            type: 'input',
-            name: 'name',
-            message: 'What the name of your token?',
-            validate: async input => {
-              if( input.search(/^[a-zA-Z0-9-_]+$/) === -1 )
-                throw new Error('Invalid characters: only use [a-zA-Z0-9-_]');
+    while (addArgument) {
+      const { argumentPlaceholder } = await inquirer.prompt([
+        {
+          type: 'input',
+          name: 'argumentPlaceholder',
+          message: 'Enter the placeholder for the argument',
+          validate: input => input.length > 0,
```

Comment:
maybe trim it as well?

Reply:
Yea these checks are somewhat older but I agree they are pretty strange. I think I might add a `isNonEmptyString` validation helper in a follow up PR 🤔 
On the other hand the goal is to help users create valid flows (or whatever they are trying to create) so a user "getting around" the validation rules is not really an issue (as long as it doesn't happen by accident and the output is valid).